qemu/tests/qemu_guest_agent.py

```diff
@@ -1341,20 +1349,25 @@ def _generate_host_keys():
             host_key = process.getoutput(cmd_get_hostkey)
             return host_key
 
-        def _login_guest_test(guest_ip):
+        def _login_guest_test(guest_user, guest_ip):
             """
             Login guest to test whether basic function works well.
 
             :param guest_ip: guest ip address.
             """
 
             cmd_login_guest = params["test_login_guest"] % guest_ip
-            login_user = process.system_output(cmd_login_guest, shell=True)
-            login_user = login_user.strip().decode(encoding="utf-8",
-                                                   errors="strict")
-            if params["guest_user"] not in login_user:
-                test.error("Can not login guest without interaction, "
-                           "basic function test is fail.")
+            output = process.system_output(cmd_login_guest, shell=True)
+            output = output.strip().decode(encoding="utf-8",
+                                           errors="strict")
+            if guest_user != 'root':
+                if guest_user not in output:
+                    test.error("Can not login guest through normal user without "
+                               "interaction, basic function test is fail.")
+            else:
+                if 'anaconda-ks.cfg' not in output:
+                    test.error("Can not login guest through user 'root' without "
+                               "interaction, basic function test is fail.")
```

Comment:
Define a param in cfg
 ```
- root:
    output_check_str = 'root'
- guest_user:
    output_check_str = 'anaconda-ks.cfg'
```

```
if params['output_check_str'] not in output:
    test.error('xxx')
```

Reply:
Done.  
(1/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_ssh_public_key_injection.root.q35: STARTED
 (1/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_ssh_public_key_injection.root.q35: PASS (55.92 s)
 (2/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_ssh_public_key_injection.non_root_user.q35: STARTED
 (2/2) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_ssh_public_key_injection.non_root_user.q35: PASS (55.22 s)